spec/unit/fuel_spec.rb

```diff
@@ -0,0 +1,80 @@
+require "spec_helper"
+
+module Wasmtime
+  RSpec.describe "Fuel" do
+    def self.test_on_store_and_caller(desc, store = :store, &block)
+      context "on Store" do
+        it desc do
+          instance_exec(send(store), &block)
+        end
+      end
+
+      context "on Caller" do
+        it desc do
+          func = Func.new(send(store), FuncType.new([], [])) do |caller|
+            instance_exec(caller, &block)
+          end
+          func.call
+        end
+      end
+    end
```

Comment:
This is a little strange but allows writing the test once for both `Store` and `Caller`. I think it's worth it.